### PR TITLE
[Packaging] Run BATS upgrade tests against the current version

### PR DESCRIPTION
**Pull request by tlrx**
BATS upgrade tests fails on master branch because it tries to install 2.x versions to upgrade from instead of 5.x versions. And since #18554 we should only test upgrades from 5.0.0-alpha4 versions.

This commit changes the vagrant tests so that it tries to list all the previous releases from version N-1. If nothing is found, it will fetch the current version and will run the upgrade tests with it. It works nicely with the current master 6.0.0-alpha1-SNAPSHOT. Once 5.0.0 is released it should run the test with it.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/UpgradableVersion.groovy [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        return canBeUpgradedTo(fromVersion(version))
+    }
+
+    boolean isStaging() {
+        return stage != null;
+    }
+
+    public boolean canBeUpgradedTo(UpgradableVersion version) {
+        // We can't upgrade from version before 5.0.0-alpha4 because of https://github.com/elastic/elasticsearch/pull/18554
+        if (this.major == 5 && this.minor == 0 && this.revision == 0 && this.stage == Stage.ALPHA && this.extra <= 4) {
```
Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/UpgradableVersion.groovy]: Really we just want to try to upgrade from the "latest" beta until we have a GA release and then we can only upgrade from GA.

Comment (nik9000) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/UpgradableVersion.groovy]: Or, better yet, we can only upgrade from the package current package until we hit a GA release.

**Comment by tlrx** [comment]
@nik9000 I removed the previous logic. Now it tries to list all the previous releases from version N-1. If nothing is found, it will fetch the current version and will run the upgrade tests with it. It works nicely with the current master 6.0.0-alpha1-SNAPSHOT. Once 5.0.0 is released it should run the test with it.

Can you please have a look and tell me what your think about this? Thanks!

File: qa/vagrant/src/test/resources/packaging/scripts/80_upgrade.bats [hunk 2]
```diff
@@ -37,6 +37,11 @@ load os_package
 # Cleans everything for the 1st execution
 setup() {
     skip_not_dpkg_or_rpm
+
+    sameVersion=
```
Comment (nik9000) [inline qa/vagrant/src/test/resources/packaging/scripts/80_upgrade.bats]: Can you make this `sameVersion=""` or `sameVersion="false"` or something like that?

**Comment by nik9000** [comment]
Awesome! Left a minor thing but otherwise LGTM.

File: qa/vagrant/src/test/resources/packaging/scripts/80_upgrade.bats (hunk 2 above)
Comment (tlrx) [inline qa/vagrant/src/test/resources/packaging/scripts/80_upgrade.bats]: Sure

**Pull request merged**